haskellPackages.{github,nix-thunk}: Fix

Ericson2314 opened the pull request:
###### Motivation for this change

`hackage-packages.nix` should be regenerated instead of hand-updated, but I don't know the proper way to do that.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Ericson2314 committed:
haskellPackages.{github,nix-thunk}: Fix

Ericson2314 commented:
Good to merge after https://github.com/obsidiansystems/nix-thunk/pull/22 too so the commit doesn't go away.

Ericson2314 commented:
@cdepillabout @sternenseemann @maralorn @expipiplus1 Know what I should do this this? This is draft only because I am not sure how to regenerated, whether this should go to `haskell-updates`, etc.

expipiplus1 commented:
This should go to haskell-updates

hackage-packages.nix can be regenerated with `./maintainers/scripts/haskell/regenerate-hackage-packages.sh --do-commit`

Thanks!

Ericson2314 committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

Ericson2314 commented:
How should I backport this to 21.11? I cherry-picked the first commit and went to regenerate, but some other things got downgraded. I guess the 21.11 packages are out of sync with the yaml :/

maralorn commented:
@Ericson2314 I am sorry. We currently haven‘t optimized the infrastructure for backports. We currently don‘t regenerate the file on stable. If possible you need to manually only cherry-pick the changes you want for the file, if you used the regenerate feature.

Ericson2314 commented:
I'll just do a regular backport then, and punt on fixing the existing issues making it cleanly regenerate. Thanks for letting me know.

cdepillabout commented on pkgs/development/haskell-modules/configuration-common.nix:
```diff
@@ -1695,12 +1695,16 @@ self: super: {
-    sha256 = "0pmx54xd7ah85y9mfi5366wbnwrp918j0wbx8yw8hrdac92qi4gh";
-  }) super.github;
+  nix-thunk = appendPatches [
+    (pkgs.fetchpatch {
+      url = "https://github.com/obsidiansystems/nix-thunk/commit/49d27a85dd39cd9413c99958c67e596756a502b5.patch";
+      sha256 = "1p1n0123yrbdqyfk4kx3gq6bdv65l1bxgbsg51ckcwclg54xp2p5";
+    })
+    (pkgs.fetchpatch {
+      url = "https://github.com/obsidiansystems/nix-thunk/commit/512867c651977265d5d8f456b538f7a364ec8a8b.patch";
+      sha256 = "121yg26y4g28k8xv7y1j6c3pxm17vsjn3vi62kkc8g928c47yd02";
+    })
```
@Ericson2314 I guess I'm a little late to this, but in the future could you make sure to include comments when you add patches like this?

If I'm a maintainer and I happen to be browsing through this file, I'll want to know the following things:

- What does the patch do?
- Why is it needed?
- When can it be deleted?
- Since you're upstream, is there some reason you can't just fix it upstream?

Ericson2314 replied:
Sorry. Those are just changing version bounds. Hopefully we make a Hackage revision soon, and then it goes away.

Ericson2314 replied:
Added in https://github.com/NixOS/nixpkgs/pull/152282

cdepillabout replied:
Thanks!

cdepillabout reviewed (commented):
It would be nice to get comments on what these patches are
